pipit/readers/hpctoolkit_reader.py

```diff
@@ -4,308 +4,1327 @@
-            # file.read(4)
+        for i in range(num_trace_headers):
+            header_pointer = trace_headers_pointer + (i * trace_header_size)
+            self.__read_single_trace_header(header_pointer)
 
-            # Trace type
-            # trace_type = int.from_bytes(
-            #     file.read(2), byteorder=byte_order, signed=signed
-            # )
-            file.read(2)
+    def __read_single_trace_header(self, header_pointer: int) -> None:
+        """
+        Reads a single trace header and all trace elements associated with it
+        """
+        self.file.seek(header_pointer)
 
-            # Offset of Trace Line start (line_ptr)
-            line_ptr = int.from_bytes(file.read(8), byteorder=byte_order, signed=signed)
+        # Index of a profile listed in the profile.db (u32)
+        profile_index = int.from_bytes(
+            self.file.read(4), byteorder=self.byte_order, signed=self.signed
+        )
+        hit = self.profile_reader.get_hit_from_profile(profile_index)
 
-            # Offset of Trace Line one-after-end (line_end)
-            line_end = int.from_bytes(file.read(8), byteorder=byte_order, signed=signed)
+        # empty space
+        self.file.read(4)
+        # Pointer to the first element of the trace line (array)
+        start_pointer = int.from_bytes(
+            self.file.read(8), byteorder=self.byte_order, signed=self.signed
+        )
+        # Pointer to the after-end element of the trace line (array)
+        end_pointer = int.from_bytes(
+            self.file.read(8), byteorder=self.byte_order, signed=self.signed
+        )
 
-            last_id = -1
+        self.file.seek(start_pointer)
 
-            last_node = None
+        # setting up some variables
+        last_id = -1  # refers to the previous context id
+        last_node: Node = None  # refers to the node associated with the last context
+        context_id: int = -1  # refers to the current context id
+        current_node: Node = (
+            None  # refers to the current node associated with the current context id
+        )
 
-            # iterate through every trace event in the process
-            for j in range(line_ptr, line_end, 12):
-                file.seek(j)
+        # refers to the least common ancestor between common_node and last_node
+        common_node: Node = None
 
-                # Timestamp (nanoseconds since epoch)
-                timestamp = (
-                    int.from_bytes(file.read(8), byteorder=byte_order, signed=signed)
-                    - min_max_time[0]
+        while self.file.tell() < end_pointer:
+            # Timestamp (nanoseconds since epoch)
+            timestamp = (
+                int.from_bytes(
+                    self.file.read(8), byteorder=self.byte_order, signed=self.signed
                 )
+                - self.min_time_stamp
+            )
+
+            # Sample calling context id (in experiment.xml)
```

Comment:
This reader no longer uses experiment.xml, correct? Probably the comment should be fixed?